packages/@react-spectrum/color/stories/ColorArea.stories.tsx

```diff
@@ -31,44 +30,81 @@ const Template: Story<SpectrumColorAreaProps> = (args) => (
   <ColorAreaExample {...args} />
 );
 
-let RGB: Set<ColorChannel> = new Set(['red', 'green', 'blue']);
 let difference = (a, b): Set<ColorChannel> => new Set([...a].filter(x => !b.has(x)));
 
 function ColorAreaExample(props: SpectrumColorAreaProps) {
   let {xChannel, yChannel, isDisabled} = props;
-  let channels = new Set([xChannel, yChannel]);
-  let zChannel: ColorChannel = difference(RGB, channels).keys().next().value;
-  let [color, setColor] = useState(props.defaultValue || parseColor('#ff00ff'));
-  return (<div role="group" aria-label="RGB Color Picker">
-    <Flex gap="size-500" alignItems="center">
-      <Flex direction="column" gap="size-50" alignItems="center">
-        <ColorArea
-          {...props}
-          value={color}
-          onChange={(e) => {
-            if (props.onChange) {
-              props.onChange(e);
+  let defaultValue = typeof props.defaultValue === 'string' ? parseColor(props.defaultValue) : props.defaultValue;
+  let [color, setColor] = useState(defaultValue || parseColor('#ff00ff'));
+  let xyChannels = new Set([xChannel, yChannel]);
+  let colorSpace = color.getColorSpace();
+  let zChannel: ColorChannel = difference(color.getColorChannels(), xyChannels).keys().next().value;
+  let isHue = zChannel === 'hue';
+
+  function onChange(e) {
+    try {
+      e = normalizeColor(e);
```

Comment:
why? doesn't onChange always give back an object?

Reply:
I'm not sure, it would appear that way given the types and from trying it out. @majornista had you done this for an explicit reason? https://github.com/adobe/react-spectrum/pull/2570/files#diff-2c7dfce7ef0c060fd764f2140018acbad9da2e7b99c81f0ae6e64fba72583033R43

Reply:
Adding `ColorField` to the story means that `parseColor` might throw an error if the string entered into the `ColorField` cannot be parsed to a color.

Reply:
I'm not seeing this happen?